### DISCUSSION TITLE
Trigger Event on hide/show any column and see the current state of the table's all columns (which are hidden/not hidden)

### DISCUSSION THREAD

**Discussion by rupom96** [KevinVandy/material-react-table#729] [Q&A]
Is there any way to trigger an event/function when hiding or showing a column? And on the trigger I want to see which columns are hidden and shown. How can I do that? Is there already any built in fn/syntax availiable for this?

**Comment by KevinVandy** [comment]
Yep easily. Manage the column visibility state, and then have a useEffect watch the column visibility state. Do what you want from there. This is shown in the state management docs.
